agent/api/serviceconnect/service_connect_attachment_parser_test.go

```diff
@@ -141,16 +143,16 @@ func getTestACSAttachments(attachmentProperties []*ecsacs.AttachmentProperty) *e
 	return &ecsacs.Attachment{
 		AttachmentArn:        strptr("attachmentArn"),
 		AttachmentProperties: attachmentProperties,
-		AttachmentType:       strptr(serviceConnectAttachmentType),
+		AttachmentType:       strptr(ServiceConnectAttachmentType),
```

Comment:
what is driving us to change encapsulation of this Type?

Reply:
Currently in `task` package when retrieving SC config we go through the list of task attachment and find the one that has `ServiceConnectAttachmentType`, before calling SC parser and validator (see [getServiceConnectConfig](https://github.com/aws/amazon-ecs-agent/blob/feature/service-connect/agent/api/task/task.go#L332)). 

I can create getter for it, or alternatively I can move the entire `getServiceConnectConfig` method from task to serviceconnect package.

Reply:
I think that aspect is very much a "task" thing more than a "serviceconnect" thing. So knowledge of the type name is probably "task" parsing a json blob or something could go in either place

Reply:
Makes sense. I'll move attachment type key back to `task`.